### ISSUE TITLE
Some Question about ffn timecost

### ISSUE THREAD

**Issue by hexisyztem** [FunAudioLLM/CosyVoice#170]
**Describe the bug**
----
I found that in the TransformerEncoder, the time consumption of the attention (attn) part is similar to that of the feed-forward network (ffn) part. However, in autoregressive models, the attn part needs to involve all time steps of the key and value in the calculations, whereas the ffn part only needs to focus on the current time step. Therefore, the time consumption of this part is relatively unusual; the time consumption of the ffn part should be significantly different from that of the attn part.
我发现 TransformerEncoder 中 attn 部分耗时跟 ffn 部分耗时接近，但是在自回归中，因为 attn 部分需要将 kv 全部时间序参与计算，但是 ffn 部分只需要关注当前时间序，因此这个部分的耗时比较异常，ffn 的耗时应当和 attn 部分存在显著差异。

**To Reproduce**
Add time logging and execute torch.cuda.synchronize() to ensure CUDA synchronization.
添加时间日志，并且执行 torch.cuda.synchronize() 确保cuda同步
```python
class TransformerEncoderLayer(nn.Module):
    """Encoder layer module.
    Args:
        size (int): Input dimension.
        self_attn (torch.nn.Module): Self-attention module instance.
            `MultiHeadedAttention` or `RelPositionMultiHeadedAttention`
            instance can be used as the argument.
        feed_forward (torch.nn.Module): Feed-forward module instance.
            `PositionwiseFeedForward`, instance can be used as the argument.
        dropout_rate (float): Dropout rate.
        normalize_before (bool):
            True: use layer_norm before each sub-block.
            False: to use layer_norm after each sub-block.
    """
    def __init__(
        self,
        size: int,
        self_attn: torch.nn.Module,
        feed_forward: torch.nn.Module,
        dropout_rate: float,
        normalize_before: bool = True,
    ):
        """Construct an EncoderLayer object."""
        super().__init__()
        self.self_attn = self_attn
        self.feed_forward = feed_forward
        self.norm1 = nn.LayerNorm(size, eps=1e-5)
        self.norm2 = nn.LayerNorm(size, eps=1e-5)
        self.dropout = nn.Dropout(dropout_rate)
        self.size = size
        self.normalize_before = normalize_before

    def forward(
        self,
        x: torch.Tensor,
        mask: torch.Tensor,
        pos_emb: torch.Tensor,
        mask_pad: torch.Tensor = torch.ones((0, 0, 0), dtype=torch.bool),
        att_cache: torch.Tensor = torch.zeros((0, 0, 0, 0)),
        cnn_cache: torch.Tensor = torch.zeros((0, 0, 0, 0)),
    ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]:
        """Compute encoded features.

        Args:
            x (torch.Tensor): (#batch, time, size)
            mask (torch.Tensor): Mask tensor for the input (#batch, time，time),
                (0, 0, 0) means fake mask.
            pos_emb (torch.Tensor): just for interface compatibility
                to ConformerEncoderLayer
            mask_pad (torch.Tensor): does not used in transformer layer,
                just for unified api with conformer.
            att_cache (torch.Tensor): Cache tensor of the KEY & VALUE
                (#batch=1, head, cache_t1, d_k * 2), head * d_k == size.
            cnn_cache (torch.Tensor): Convolution cache in conformer layer
                (#batch=1, size, cache_t2), not used here, it's for interface
                compatibility to ConformerEncoderLayer.
        Returns:
            torch.Tensor: Output tensor (#batch, time, size).
            torch.Tensor: Mask tensor (#batch, time, time).
            torch.Tensor: att_cache tensor,
                (#batch=1, head, cache_t1 + time, d_k * 2).
            torch.Tensor: cnn_cahce tensor (#batch=1, size, cache_t2).
        """
        torch.cuda.synchronize() # cuda tongbu
        time1 = time.perf_counter()
        residual = x
        if self.normalize_before:
            x = self.norm1(x)
        x_att, new_att_cache = self.self_attn(x, x, x, mask, pos_emb=pos_emb, cache=att_cache)

        x = residual + self.dropout(x_att)
        if not self.normalize_before:
            x = self.norm1(x)

        residual = x
        if self.normalize_before:
            x = self.norm2(x)
        torch.cuda.synchronize()
        time2 = time.perf_counter()
        x = residual + self.dropout(self.feed_forward(x))
        if not self.normalize_before:
            x = self.norm2(x)

        torch.cuda.synchronize()
        time3 = time.perf_counter() #benchmark

        print(x.shape, att_cache.shape)
        print("time cost: ", time2-time1, time3-time2)

        fake_cnn_cache = torch.zeros((0, 0, 0), dtype=x.dtype, device=x.device)
        return x, mask, new_att_cache, fake_cnn_cache
```

**Comment by aluminumbox** [comment]
during inference, there is att_cache, I believe the ffn layer will not recalculate the history part, you can check the detail in self.self_attn(x, x, x, mask, pos_emb=pos_emb, cache=att_cache)
